### PR TITLE
#37 Add popup for editing TO DO item

### PR DESCRIPTION
**Pull request by lizaveta-karnila**

### CODE REVIEW

File: src/components/Card.jsx [hunk 1]
```diff
@@ -1,19 +1,28 @@
-import React, { useContext } from 'react';
+import React, { useState, useContext } from 'react';
 import { ThemeContext } from '../contexts/theme-context';
+import Popup from './Popup/Popup';
 
-export function Card({ card, index, markCard, deleteCard }) {
+export function Card({ card, index, markCard, renameCard, deleteCard }) {
```
Comment (siarhei-hrynevich) [inline src/components/Card.jsx]: I think 1 callback that updates whole card better than few callbacks for each field

File: src/components/Popup/popup.css [hunk 2]
```diff
@@ -0,0 +1,85 @@
+.popup {
```
Comment (siarhei-hrynevich) [inline src/components/Popup/popup.css]: Looks like overlay

File: src/components/Popup/Popup.jsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import React, { useState } from "react";
+import "./popup.css";
+import TextareaAutosize from 'react-textarea-autosize';
+
+function Popup({ trigger, setTrigger, card, updateCard }) {
+  const [name, setName] = useState(card.name);
+  const [isDone, setIsDone] = useState(card.isDone);
```
Comment (siarhei-hrynevich) [inline src/components/Popup/Popup.jsx]: Why didn't you aggregate both fields in object?

File: src/components/Popup/Popup.jsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import React, { useState } from "react";
+import "./popup.css";
+import TextareaAutosize from 'react-textarea-autosize';
+
+function Popup({ trigger, setTrigger, card, updateCard }) {
```
Comment (siarhei-hrynevich) [inline src/components/Popup/Popup.jsx]: I prefer to make Popup component that will represent empty popup that will serve content from props.children.
And in thin case you should create something like EditCardForm component that will be served

File: src/components/Card.jsx [hunk 5]
```diff
@@ -1,19 +1,28 @@
...
+    <>
+      <div className='card' onClick={() => { setPopupIsOpen(true) }} style={{ backgroundColor: card.isDone ? theme.cardBgGreen : theme.cardBgRed }}>
+        <div className='card__text' style={{ color: theme.textColor }}>{card.name}</div>
+        <div className='card__delete-btn' onClick={(evt) => { evt.stopPropagation(); deleteCard(index) }} style={{ backgroundColor: theme.delBtnColor, color: theme.delBtnTextColor }}>+</div>
+        <div className='card__status' style={{ color: theme.textColor }}>{card.isDone ? 'completed' : 'not completed'}</div>
+        <div className="card_timestamp" style={{ color: theme.textColor }}> {(card.timestamp)}</div>
+      </div>
+
+      <Popup trigger={popupIsOpen} setTrigger={setPopupIsOpen} card={card} updateCard={updateCard}>
+        <h3>My popup</h3>
```
Comment (siarhei-hrynevich) [inline src/components/Card.jsx]: It is used in Popup component?

**Review by siarhei-hrynevich — CHANGES_REQUESTED** [review]

File: src/components/Card.jsx (hunk 5 above)
Comment (lizaveta-karnila) [reply src/components/Card.jsx]: no :)

File: src/components/Popup/popup.css (hunk 2 above)
Comment (lizaveta-karnila) [reply src/components/Popup/popup.css]: done

File: src/components/Card.jsx (hunk 1 above)
Comment (lizaveta-karnila) [reply src/components/Card.jsx]: done

File: src/components/Popup/Popup.jsx (hunk 3 above)
Comment (lizaveta-karnila) [reply src/components/Popup/Popup.jsx]: done

File: src/components/Popup/Popup.jsx (hunk 4 above)
Comment (lizaveta-karnila) [reply src/components/Popup/Popup.jsx]: done

File: src/components/App.jsx [hunk 6]
```diff
@@ -25,15 +25,8 @@ export function App() {
...
-    setCards([...cards, { id: uuidv4(), text: text, isDone: false, timestamp: new Date() }]);
-  };
-
-  const markCard = index => {
-    setCards(cards.map((item, i) => {
-      if (i === index) item.isDone = !item.isDone;
-      return item;
-    }))
+  const addCard = name => {
+    setCards([...cards, { id: uuidv4(), name: name, isDone: false, timestamp: new Date() }]);
```
Comment (siarhei-hrynevich) [inline src/components/App.jsx]: Create constuctor for item

File: src/components/EditCardForm/EditCardForm.jsx [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+  const handleInputChange = event => {
+    const target = event.target;
+    const value = target.type === 'checkbox' ? target.checked : target.value;
+    const name = target.name;
+    setNewCard({ ...newCard, [name]: value });
+  }
+
+  const handleSubmit = event => {
+    updateOpenCard(newCard.name, newCard.isDone);
```
Comment (siarhei-hrynevich) [inline src/components/EditCardForm/EditCardForm.jsx]: updateOpenCard shold has 1 params (object) that contains all fields

**Review by siarhei-hrynevich — CHANGES_REQUESTED** [review]

File: src/components/EditCardForm/EditCardForm.jsx (hunk 7 above)
Comment (lizaveta-karnila) [reply src/components/EditCardForm/EditCardForm.jsx]: done

File: src/components/App.jsx (hunk 6 above)
Comment (lizaveta-karnila) [reply src/components/App.jsx]: done

**Pull request merged**